### DISCUSSION TITLE
For Logseq, How to modify note template to make the image showed up properly in Zotero notes when sync?

### DISCUSSION THREAD

**Discussion by chs33** [windingwind/zotero-better-notes#774] [Q&A] [answered]
in Logseq, the image can be inserted through following syntax: ![](some_relative_path), which will make BN complain when sync the notes back to Zotero.
![image](https://github.com/windingwind/zotero-better-notes/assets/16418270/3e091969-ff0d-4c7b-b5d3-47f221a219d0)
is shown as 
![image](https://github.com/windingwind/zotero-better-notes/assets/16418270/97accb1c-3cf8-4b22-a38c-ae12ee300429)
in Logseq.

It will fail when sync this to the Zotero through BN
![image](https://github.com/windingwind/zotero-better-notes/assets/16418270/9c47dd68-8bbd-4fa3-b2c2-c649f215cbbe)

In general seems like most of the md rendering/workflow is designed working for md notes in Obsidian. More support Logseq in the future is very much appreciated!

**Comment by windingwind** [comment]
I'll see. But I guess there is not much thing we can do with the attributes (width & height). I'll try to fix the Path problem.

**Reply by windingwind** [reply, marked as answer]
fixed.

**Reply by chs33** [reply]
Thanks for the quick fix. When BN sync back to the markdown notes, Looks like it will change the original syntax which is not desired:

before sync:
![image](https://github.com/windingwind/zotero-better-notes/assets/16418270/5eee3864-e739-4f70-bd35-bbfc7b9f972b)
after sync
![image](https://github.com/windingwind/zotero-better-notes/assets/16418270/b911956d-fb59-4c03-9a42-379d002b5a32)

For the attributes (width & height) I am curious the reason you said there is not too much you can do, is it because it is third party module that conduct the md to html (as well as the other way around) so that BN don't have too much freedom on that part?

**Reply by windingwind** [reply]
The image `alt` is used for storing Zotero image information, which avoids importing duplicate images while syncing. Generally, this won't affect how your markdown is rendered.

The height and width you use here is not a standard markdown syntax, thus I do not plan to add support for this accent. There are tons of markdown accents in the world, we cannot support them all. As long as BN does not break the original markdown, I don't think we should take any action.

**Reply by chs33** [reply]
In this case it does make the rendering in Logseq fail.
![image](https://github.com/windingwind/zotero-better-notes/assets/16418270/30381c1f-a3c4-41c0-a8e0-8771272bf457)

**Reply by windingwind** [reply]
Could you please check if the image exists under the new path (`attachments.MB....png`)? Seems like the image is missing.

**Reply by chs33** [reply]
it is there. 

![image](https://github.com/windingwind/zotero-better-notes/assets/16418270/21d1f38c-e515-41c4-bc98-d5243b9e99e6)

![image](https://github.com/windingwind/zotero-better-notes/assets/16418270/e855a5b8-21b6-4e79-8980-ba7ccda6b87b)

![image](https://github.com/windingwind/zotero-better-notes/assets/16418270/af25b662-2e1a-47d8-b56d-9b0836f70152)

I suppose it is more like a problem for Logseq rendering?

TBH, I would very much prefer Zotero just load the original pic at original location anyway, and when output to markdown keep the original image inserting syntax(the standard part), that is not feasible?

**Reply by windingwind** [reply]
Seems like logseq does not recognize the image path. Try `./attachment/..png`, does that work? I have to complain that these markdown software do that many fancy features, but fail with such basic, simple, and important image path recognition. Obsidian cannot recognize HTML images, and now logseq cannot recognize relative paths...

> Zotero just load the original pic at original location

Not possible, for security reasons and for cross-platform syncing reasons. Zotero always import images and store them for syncing.

> Output to markdown keep the original image inserting syntax

That's probably possible, but I haven't had any idea of how to do that. I'll try.

**Reply by chs33** [reply]
>  Try ./attachment/..png, does that work?

No it doesn't : )

> That's probably possible, but I haven't had any idea of how to do that. I'll try.

Very much appreciated!

**Reply by windingwind** [reply]
According to GPT: [According to the web search results, Logseq does not support relative image paths for images that are not in the ./assets directory](https://discuss.logseq.com/t/relative-path-for-zotero-data-directory/4568)[1](https://discuss.logseq.com/t/relative-path-for-zotero-data-directory/4568)[2](https://github.com/logseq/logseq/issues/4181). This means that if you want to insert an image from a different folder, you have to use the absolute path, which may not work across different devices or platforms. [Some users have requested this feature or suggested some workarounds](https://bing.com/search?q=logseq+relative+image+path)[3](https://bing.com/search?q=logseq+relative+image+path)[4](https://discuss.logseq.com/t/relative-paths-in-file-links-dont-work-incorrect-handling-of-double-dot/8952)[5](https://discuss.logseq.com/t/images-that-are-not-in-the-assets-dir-should-also-be-exported/14229), but it seems that there is no official solution yet. I hope this answers your question. 😊

Could you please test if an image like `./assets/...png` would work? If so, I can add an option to allow users to configure the output attachment folder.

**Reply by chs33** [reply]
it works.
![image](https://github.com/windingwind/zotero-better-notes/assets/16418270/6834ff94-7b4d-4c36-8ce9-92e3aa6e6158)

It is the case that only image in assets is recognized.

**Reply by windingwind** [reply]
Alright. Strange design. I'll push a new release today to allow customizing output attachment folder.

**Reply by windingwind** [reply]
https://github.com/windingwind/zotero-better-notes/releases/tag/1.1.4-34

**Reply by chs33** [reply]
Thanks. The path must be absolute?

**Reply by windingwind** [reply]
no, it's the relative path (or, folder name), based on the markdown file directory. i haven't supported abs path, but we can do that.

**Reply by chs33** [reply]
Definite prefer relative path but somehow relative path doesn't work for me 
![image](https://github.com/windingwind/zotero-better-notes/assets/16418270/6870712d-9ebe-4c4a-b191-75f6efdd099e)
![image](https://github.com/windingwind/zotero-better-notes/assets/16418270/765fe7e0-6e1a-4683-b1de-2f1af477d856)

**Reply by windingwind** [reply]
please try 1.1.4-35
